src/components/editor/Bindings/FieldSelection/index.tsx

```diff
@@ -343,15 +322,54 @@ function FieldSelectionViewer({ collectionName }: Props) {
         setSelectionSaving,
     ]);
 
+    const updateTableSettings = (
+        event: SyntheticEvent,
+        checked: boolean,
+        column: string
+    ) => {
+        event.preventDefault();
+        event.stopPropagation();
+
+        const existingSettings = tableSettings ?? {};
+
+        const shownOptionalColumns = Object.hasOwn(
+            existingSettings,
+            TablePrefixes.fieldSelection
+        )
+            ? existingSettings[TablePrefixes.fieldSelection]
+                  .shownOptionalColumns
+            : [];
+
+        const evaluatedSettings =
+            !checked && shownOptionalColumns.includes(column)
+                ? {
+                      ...existingSettings,
+                      [TablePrefixes.fieldSelection]: {
+                          shownOptionalColumns: shownOptionalColumns.filter(
+                              (value) => value !== column
+                          ),
+                      },
+                  }
+                : checked && !shownOptionalColumns.includes(column)
+                ? {
+                      ...existingSettings,
+                      [TablePrefixes.fieldSelection]: {
+                          shownOptionalColumns: [
+                              ...shownOptionalColumns,
+                              column,
+                          ],
+                      },
+                  }
+                : existingSettings;
```

Comment:
Small tweak here. Even though the list isn't huge I think we should store off `shownOptionalColumns.includes(column)` into a variable so we aren't figuring that out twice.